src/skill.cpp

```diff
@@ -217,6 +219,88 @@ bool SkillLevel::can_train() const
     return get_option<float>( "SKILL_TRAINING_SPEED" ) > 0.0;
 }
 
+const SkillLevel &SkillLevelMap::get_skill_level_object( const skill_id &ident ) const
+{
+    static const SkillLevel null_skill;
+
+    if( ident && ident->is_contextual_skill() ) {
+        debugmsg( "Skill \"%s\" is context-dependent. It cannot be assigned.", ident.str() );
+        return null_skill;
+    }
+
+    const auto iter = find( ident );
+
+    if( iter != end() ) {
+        return iter->second;
+    }
+
+    return null_skill;
+}
+
+SkillLevel &SkillLevelMap::get_skill_level_object( const skill_id &ident )
+{
+    static SkillLevel null_skill;
+
+    if( ident && ident->is_contextual_skill() ) {
+        debugmsg( "Skill \"%s\" is context-dependent. It cannot be assigned.", ident.str() );
+        return null_skill;
+    }
+
+    return ( *this )[ident];
+}
+
+void SkillLevelMap::mod_skill_level( const skill_id &ident, int delta )
+{
+    SkillLevel &obj = get_skill_level_object( ident );
+    obj.level( obj.level() + delta );
+}
+
+int SkillLevelMap::get_skill_level( const skill_id &ident ) const
+{
+    return get_skill_level_object( ident ).level();
+}
+
+int SkillLevelMap::get_skill_level( const skill_id &ident, const item &context ) const
+{
+    auto id = context.is_null() ? ident : context.contextualize_skill( ident );
+    return get_skill_level( std::move( id ) );
```

Comment:
The move here is useless, the other `get_skill_level` function takes its argument as reference anyway.

Reply:
True.  It's also harmless, and might help in the future if that function changes.  But I can see that's unlikely, so happy to change if you wish.

Reply:
On reflection, it has a performance impact on debug builds, so I'll get rid of it.